### ISSUE TITLE
[Android] onValueChange stopped firing event for false values (null, undefined, 0, "", false, etc)

### ISSUE THREAD

**Issue by MaxTibs** [react-native-picker/picker#590]
Library version: 2.8.1
React native version: 0.74.6

After upgrading from 7.6.2 to 8.2.0, the `onValueChange` handler stopped being called for false values (false, "", undefined, null, etc). 
It seems it also stopped firing event on placeholder touch event. See this [thread](https://github.com/lawnstarter/react-native-picker-select/issues/604).

After some research I found where the breaking change has been introduced: [here](https://github.com/react-native-picker/picker/commit/243ead2e226ea5441a7091beccc903e804922c18#diff-3f055065a2b6ec0273bd0a4500979cb49e13dee44504e0d2a3015d8d312637d4L203)

IMO it's should be up to the developer to interpret the value returned by the picker. If null is meant to represent the placeholder value, then it should be the developer responsibility to handle that null value from it's onValueChange handler. I don't think the values should be muted from within the picker implementation.

In other words, this change should probably be reverted

**Comment by dmk3141618** [comment]
Same issue on Android.

I can set empty string as a value on the iOS device.

But I can not set empty string as a value on the Android device. It does not trigger onValueChange.
value is '', 'option1', 'option2'
then '' >> 'option1' , it triggers onValueChange.
then 'option1' >> 'option2' , it triggers onValueChange.
But 'option2' >> '' , it does not trigger onValueChange.
So I can not reset value to the empty state.

on the iOS device, it works.

**Comment by dmk3141618** [comment]
Any updates? Sorry but I really need this fixed.

**Comment by didley** [comment]
@dmk3141618 temporary fix, if you haven't tired, is downgrading to `2.7.6`.

**Comment by dmk3141618** [comment]
Is there any update?

**Comment by dmk3141618** [comment]
This PR works. When is going to be applied to the real version?
fix: setting nullish values on android #604
if (value !== undefined) {

**Comment by didley** [comment]
Sorry @Naturalclar, any chance #604 can be reviewed?

**Comment by dmk3141618** [comment]
Thank you kindly.
